internal/patroni/api.go

```diff
@@ -177,3 +184,39 @@ func (exec Executor) RestartPendingMembers(ctx context.Context, role, scope stri
 
 	return err
 }
+
+// GetTimeline gets the patronictl status and returns the timeline,
+// currently the only information required by PGO.
+// Returns `0` as the int32 if it runs into errors or cannot find
+// a running Leader pod to get the up-to-date timeline from
+func (exec Executor) GetTimeline(ctx context.Context) (int32, error) {
+	var stdout, stderr bytes.Buffer
+
+	// The following exits zero when it is able to read the DCS and communicate
+	// with the Patroni HTTP API. It prints the result of calling "GET /cluster"
+	// - https://github.com/zalando/patroni/blob/v2.1.1/patroni/ctl.py#L849
+	err := exec(ctx, nil, &stdout, &stderr,
+		"patronictl", "list", "-f", "json")
```

Comment:
⛏️ I prefer long arguments for clarity.

```suggestion
		"patronictl", "list", "--format", "json")
```